### PR TITLE
Global config

### PR DESCRIPTION
**Pull request by samarthsharma36**
-> Tests for all Module Types covered
-> Styling changed
@yvesh  @puneet0191 
You need to pull this  PR in order to test 'Modules ' PR

### CODE REVIEW

File: src/JoomlaBrowser.php [hunk 1]
```diff
@@ -1121,4 +1121,177 @@ public function createCategory($title, $extension = '')
...
+		$this->debug('I set the site to offline');
+		$this->doAdministratorLogin();
+		$this->amOnPage('/administrator/index.php?option=com_config');
+		if ($enable == true){
+			$this->click(['id' => 'jform_offline1']);
+		}else{
+			$this->click(['id' => 'jform_offline0']);
+		}
+		$this->click("Save");
+		$this->see('Configuration saved', ['id' => 'system-message-container']);
```
Comment (nhungredweb) [inline src/JoomlaBrowser.php:1078]: the ID or content will be better if you move it on Locator page . cc @Samgithub1997

File: src/JoomlaBrowser.php [hunk 2]
```diff
@@ -1121,4 +1121,177 @@ public function createCategory($title, $extension = '')
...
+	 *
+	 * @param   \AcceptanceTester   $I  Acceptance Tester
+	 *
+	 * @return void
+	 */
+	public function setSiteSearchEngineFriendly($enable == true)
+	{
+		$this->debug('I set the Search engine optimisation to Yes');
+		$this->doAdministratorLogin();
+		$this->amOnPage('/administrator/index.php?option=com_config');
```
Comment (nhungredweb) [inline src/JoomlaBrowser.php:1092]: push the url on Locator will be better .

**Comment by Samgithub1997** [comment]
@nhungredweb I have added URL to Locator and used them in createModule function.
Please review the changes :)

**Comment by samarthsharma36** [comment]
@nhungredweb @yvesh @puneet0191  I have added the suggested changes

File: src/JoomlaBrowser.php [hunk 3]
```diff
@@ -744,12 +700,10 @@ public function searchForItem($name = null)
 		if ($name)
 		{
 			$this->debug("Searching for $name");
-			$this->fillField(['id' => "filter_search"], $name);
+			$this->fillField(['class' => "form-control"], $name);
```
Comment (puneet0191) [inline src/JoomlaBrowser.php:703]: why is this change needed?

File: src/JoomlaBrowser.php [hunk 4]
```diff
@@ -1099,26 +1042,195 @@ public function createCategory($title, $extension = '')
...
 		$this->see('Category saved', ['id' => 'system-message-container']);
 		$this->checkForPhpNoticesOrWarnings();
 	}
+
+	/**
+	 * Site Offline
+	 *
+	 * @return  void
+	 *
+	 * @since   3.7.5
```
Comment (puneet0191) [inline src/JoomlaBrowser.php:1065]: wrong Since, remember these changes are for 4.0 Version, and your PR is also on the 4.0 branch

File: src/JoomlaBrowser.php (hunk 2 above)
Comment (puneet0191) [reply src/JoomlaBrowser.php:1092]: only put the URL's on Page Class when it is being used at many places, if it is just for one function, I think it is fine to keep it directly.

File: src/JoomlaBrowser.php [hunk 5]
```diff
@@ -1099,26 +1042,195 @@ public function createCategory($title, $extension = '')
...
+		}
+		else
+		{
+			$this->click(['id' => 'jform_sef0']);
+		}
+		$this->click("Save");
+		$this->see('Configuration saved', ['id' => 'system-message-container']);
+	}
+
+	/**
```
Comment (puneet0191) [inline src/JoomlaBrowser.php:1107]: @Samgithub1997  have you checked style here? I don't think it is according to the way we discussed.

File: src/JoomlaBrowser.php [hunk 6]
```diff
@@ -1099,26 +1042,195 @@ public function createCategory($title, $extension = '')
...
+        // Title
+        $this->fillField(Locators::$moduleTitle, $moduleName);
+        /**
+         * Set Position
+         * setModulePosition not working. Tried both
+         * $I->setModulePosition($moduleName,'Main Top[main-top]'); and
+         * $I->setModulePosition($moduleName,'Main Top');
+         */
+        $this->click('Type or Select a Position');
+        $this->fillField(Locators::$modulePosition,'Main Top');
```
Comment (puneet0191) [inline src/JoomlaBrowser.php:1134]: leave a space before the second parameter
`$this->fillField(Locators::$modulePosition, 'Main Top');`

**Review by puneet0191 — DISMISSED** [review]

**Review by puneet0191 — CHANGES_REQUESTED** [review]
@Samgithub1997 could you update the styling changes here as well

File: src/JoomlaBrowser.php (hunk 3 above)
Comment (samarthsharma36) [reply src/JoomlaBrowser.php:703]: id wasn't working in mine. So I changed it to class

**Comment by samarthsharma36** [comment]
Sorry, I forgot to close this branch. The pull request #172 is the main one. I will make the above suggested changes @puneet0191 @nhungredweb
